src/crimson/osd/heartbeat.cc

```diff
@@ -310,19 +310,21 @@ seastar::future<> Heartbeat::maybe_share_osdmap(
   }
   auto& peer = found->second;
 
-  if (m->map_epoch > peer.get_projected_epoch()) {
-    logger().debug("{} updating peer {} session's projected_epoch"
-                   "from {} to ping map epoch of {}",
-                   __func__, from, peer.get_projected_epoch(),
-                   m->map_epoch);
-    peer.set_projected_epoch(m->map_epoch);
```

Comment:
We want to update projected_epoch if m->map_epoch is newer, even if it turns out that we don't need to send a map.  Remember, there are many OSDs that might have sent maps to this peer, not just us.